explorer/common/error_builders.h

```diff
@@ -33,6 +33,10 @@ inline auto RuntimeError(SourceLocation loc) -> ErrorBuilder {
   return ErrorBuilder("RUNTIME ERROR", loc.ToString());
 }
 
+inline auto AssertionError(SourceLocation loc) -> ErrorBuilder {
```

Comment:
I'm not sure if we need a new kind of error for this -- it seems like `RuntimeError` would work.

Reply:
Yes RuntimeError would work. But I think RuntimeError is sent by the interpreter that something is wrong. With asserts it decides the user and that should be distinguished, or not ?

Reply:
I believe `RuntimeError` is for run-time bugs in user code, and we use `CHECK` for bugs in the interpreter itself. An assertion failure indicates a bug in user code, so `RuntimeError` seems like a good fit.

Reply:
OK...!

Reply:
Done